src/service/url-replacements-impl.js

```diff
@@ -454,7 +454,7 @@ export class GlobalVariableSource extends VariableSource {
     this.set('AMP_VERSION', () => '$internalRuntimeVersion$');
 
     this.set('BACKGROUND_STATE', () => {
-      return viewerForDoc(this.ampdoc.win.document).isVisible() ? '0' : '1';
+      return viewerForDoc(this.ampdoc).isVisible() ? '0' : '1';
```

Comment:
The old code is "ugh". I'm thinking maybe we should consider prohibiting doc-service lookup via `Document` argument to avoid this kind of silly bugs.